src/templates/partials/lighttpd.hbs

```diff
@@ -11,36 +15,59 @@ $SERVER["socket"] == ":80" {
-    ssl.pemfile               = "/path/to/signed_cert_plus_private_key"
-    ssl.ca-file               = "/path/to/intermediate_certificate"
+    ssl.pemfile = "/path/to/signed_cert_plus_private_key"
+    ssl.ca-file = "/path/to/intermediate_certificate"
+  {{/if}}
 {{#if output.usesDhe}}
     {{#if (minver "1.4.29" form.serverVersion)}}
 
     # {{output.dhCommand}} > /path/to/dhparam
-    ssl.dh-file               = "/path/to/dhparam"
+    ssl.dh-file = "/path/to/dhparam"
     {{/if}}
 {{/if}}
 
     # {{form.config}} configuration
-{{#if (minver "1.4.48" form.serverVersion)}}
-    ssl.openssl.ssl-conf-cmd = ("Protocol" => "ALL, -SSLv2, -SSLv3{{#unless (includes "TLSv1" output.protocols)}}, -TLSv1{{/unless}}{{#unless (includes "TLSv1.1" output.protocols)}}, -TLSv1.1{{/unless}}{{#unless (includes "TLSv1.2" output.protocols)}}, -TLSv1.2{{/unless}}")
-{{else}}
-    # Please upgrade to 1.4.48 or else you cannot fully disable deprecated protocols
+ {{#if (minver "1.1.0" form.opensslVersion)}}
+  {{#if (minver "1.4.55" form.serverVersion)}}
+    ssl.openssl.ssl-conf-cmd = ("MinProtocol" => {{#if (includes "TLSv1" output.protocols)}}"TLSv1"{{else if (includes "TLSv1.1" output.protocols)}}"TLSv1.1"{{else if (includes "TLSv1.2" output.protocols)}}"TLSv1.2"{{else}}"TLSv1.3"{{/if}}, "Options" => "-SessionTicket")
+  {{else if (minver "1.4.48" form.serverVersion)}}
+    ssl.openssl.ssl-conf-cmd = ("Protocol" => "ALL, -SSLv2, -SSLv3{{#unless (includes "TLSv1" output.protocols)}}, -TLSv1{{/unless}}{{#unless (includes "TLSv1.1" output.protocols)}}, -TLSv1.1{{/unless}}{{#unless (includes "TLSv1.2" output.protocols)}}, -TLSv1.2{{/unless}}", "Options" => "-SessionTicket")
+  {{else}}
+    # Please upgrade to at least lighttpd 1.4.48 or else you can not fully disable deprecated protocols
     ssl.use-sslv2 = "disable"
     ssl.use-sslv3 = "disable"
-{{/if}}
-    ssl.cipher-list           = "{{{join output.ciphers ":"}}}"
-    ssl.honor-cipher-order    = "{{#if output.serverPreferredOrder}}enable{{else}}disable{{/if}}"
-{{#if form.hsts}}
-
-    # HTTP Strict Transport Security ({{output.hstsMaxAge}} seconds)
-    setenv.add-response-header  = (
-        "Strict-Transport-Security" => "max-age={{output.hstsMaxAge}}"
-    )
-{{/if}}
-}
\ No newline at end of file
+  {{/if}}
+ {{else}}
+    # Please upgrade to openssl 1.1.1 or later
+    # Earlier openssl versions no longer receive security patches from openssl.org
+    ssl.use-sslv2 = "disable"
+    ssl.use-sslv3 = "disable"
+ {{/if}}
+    ssl.cipher-list = "{{{join output.ciphers ":"}}}"
+    # ssl.honor-cipher-order = "disable" respects client cipher preference
+    # (safe when ssl.cipher-list is restricted to only ciphers which support Perfect Forward Secrecy)
+    ssl.honor-cipher-order = "{{#if output.serverPreferredOrder}}enable{{else}}disable{{/if}}"
+}
+#$SERVER["socket"] == "[::]:443" { ... } # repeat entire $SERVER["socket"] == ":443" { ... } config above for IPv6
```

Comment:
Why is it not possible to have a config that works for both IPv4 and IPv6? I find it hard to believe it requires you to repeat all of your config twice.

Reply:
Sorry to disappoint, but it does.  To avoid repetition, you can use the 'include' directive to include a separate file in each condition, and the separate file lists the ssl config once.

This is addressed a bit better in lighttpd 1.4.56.

Reply:
(See #122 for how this is handled in lighttpd 1.4.56)

FYI: lighttpd 1.4.56 is on track to be released next weekend, so I am hoping you have a chance to review #122 soon, too!  Thanks!